packages/kbn-interpreter/src/common/interpreter/interpreter_provider.js

```diff
@@ -33,7 +33,7 @@ import { interpretProvider } from './interpret';
   socket: the socket to communicate over
 */
 
-export function socketInterpreterProvider({
+export function interpreterProvider({
   types,
   functions,
   handlers,
```

Comment:
I think socket param here can be removed.

Reply:
Yeah, it looks like there's some other updates needed in here. `onFunctionNotFound`, which we used to use to handle the fallthrough to other runtimes, still has some socket stuff in it.

In `packages/kbn-interpreter/src/common/interpreter/interpret.js`, there's a check for a function definition not existing, which still falls through to the `onFunctionNotFound` handler... is that a scenario that is even possible anymore? I haven't spent a lot of time with this PR, but I thought now that most things run in the browser that the browser knows if a function exists before that point, and can bail earlier.

Reply:
Great points. I was trying to keep this a minimal PR, but it looks like I missed quite a lot... A bit sloppy of me.